packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:vm_service/vm_service.dart';
+
+import '../../../../../shared/globals.dart';
+import '../../../shared/primitives/class_name.dart';
+import '../../../shared/primitives/instance_set_view.dart';
+
+class HeapClassSampler extends ClassSampler {
+  HeapClassSampler(this.className);
+
+  final HeapClassName className;
+
+  @override
+  Future<void> oneVariableToConsole() async {
+    final isolateRef = serviceManager.isolateManager.mainIsolate.value!;
+    final isolateId = isolateRef.id!;
+
+    // It would be great to find out how to avoid full scan of classes.
```

Comment:
should this be a TODO and an issue?

Reply:
I am not sure it is possible. So, just TODO to make reader aware.